pmd-scala-modules/pmd-scala-common/src/main/java/net/sourceforge/pmd/cpd/ScalaTokenAdapter.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.cpd;
+
+import net.sourceforge.pmd.lang.ast.GenericToken;
+
+import scala.meta.tokens.Token;
+
+/**
+ * Adapts the scala.meta.tokens.Token so that it can be used with the generic BaseTokenFilter
+ */
+public class ScalaTokenAdapter implements GenericToken {
+
+    private Token token;
+    private GenericToken previousComment;
+
+    ScalaTokenAdapter(Token token, GenericToken comment) {
+        this.token = token;
+        this.previousComment = comment;
+    }
+
+    @Override
+    public GenericToken getNext() {
+        // not required
+        return null;
```

Comment:
Please either implement this or throw `NotImplementedException` (from lang3) or `UnsupportedOperationException`

Reply:
Would also be nice to add a toString to this class (any autogenerated one will do)

Reply:
now throws